src/main/java/org/broadinstitute/hellbender/tools/HaplotypeCallerSpark.java

```diff
@@ -114,16 +122,21 @@ public boolean requiresReference(){
 
     @Override
     protected void runTool(final JavaSparkContext ctx) {
-        final List<SimpleInterval> intervals = hasIntervals() ? getIntervals() : IntervalUtils.getAllIntervalsForReference(getHeaderForReads().getSequenceDictionary());
-        final JavaRDD<VariantContext> variants = callVariantsWithHaplotypeCaller(getAuthHolder(), ctx, getReads(), getHeaderForReads(), getReference(), intervals, hcArgs, shardingArgs);
+        // Reads must be coordinate sorted to use the overlaps partitioner
+        final SAMFileHeader readsHeader = getHeaderForReads().clone();
+        readsHeader.setSortOrder(SAMFileHeader.SortOrder.coordinate);
```

Comment:
Yes, this makes absolutely sure it's sorted - there may be an issue with sharded input not actually being sorted, even though the headers says it is. There's a follow-up issue to check and document sorting requirements: #3308.